edgecast/resources/cps/certificate.go

```diff
@@ -422,9 +422,53 @@ func ResourceCertificateRead(ctx context.Context,
 	d *schema.ResourceData,
 	m interface{},
 ) diag.Diagnostics {
-	// Not Yet Implemented
-	var diags diag.Diagnostics
-	return diags
+	config := m.(**api.ClientConfig)
+	cpsService, err := buildCPSService(**config)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	certID, _ := strconv.ParseInt(d.Id(), 10, 64)
+
+	log.Printf(
+		"[INFO] Retriving certificate : ID: %v",
+		certID,
+	)
+
+	params := certificate.NewCertificateGetParams()
+	params.ID = certID
+	resp, err := cpsService.Certificate.CertificateGet(params)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[INFO] Retrieved certificate: %# v", pretty.Formatter(resp))
+
+	d.SetId(strconv.Itoa(int(resp.ID)))
```

Comment:
Github did a weird thing so I resolved the thread and commenting again. I believe we do not want to set the ID in the read here since we do not expect it to change.

Reply:
Thanks for catching this! Updated.